src/viewer.js

```diff
@@ -373,6 +376,19 @@ $.Viewer = function( options ) {
         if (!_this._updateRequestId) {
             _this._updateRequestId = scheduleUpdate( _this, updateMulti );
         }
+
+        // Attaches a handler to monitor individual TiledImage loading states
+        // propagate aggregate viewer-level fully-loaded status changes.
+        var tiledImage = event.item;
+        tiledImage.addHandler('fully-loaded-change', function() {
+            // Check aggregate loading state across all TiledImages
+            var newFullyLoaded = self.areAllFullyLoaded();
+            // Only trigger event if viewer's loading state has changed
+            if (newFullyLoaded !== self._fullyLoaded) {
+                self._fullyLoaded = newFullyLoaded;
+                self.raiseEvent('fully-loaded-change', { fullyLoaded: newFullyLoaded });
```

Comment:
This event needs doc comments like we do with other events. You can base it off the doc comments for the `TiledImage` `fully-loaded-change` event.